### PR TITLE
[SPARK-14544] [SQL] improve performance of SQL UI tab

### PR DESCRIPTION
**Pull request by davies**
## What changes were proposed in this pull request?

This PR improve the performance of SQL UI by:

1) remove the details column in all executions page (the first page in SQL tab). We can check the details by enter the execution page.
2) break-all is super slow in Chrome recently, so switch to break-word.
3) Using "display: none" to hide a block.
4) using one js closure for  for all the executions, not one for each.
5) remove the height limitation of details, don't need to scroll it in the tiny window.
## How was this patch tested?

Exists tests.

ui

### CODE REVIEW

**Comment by davies** [comment]
cc @zsxwing

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala [hunk 1]
```diff
@@ -55,6 +55,12 @@ private[ui] class AllExecutionsPage(parent: SQLTab) extends WebUIPage("") with L
       }
       _content
     }
+    content ++=
+      <script>
+        function clickDetail(details) {{
+          details.parentNode.querySelector('.stage-details').classList.toggle('collapsed')
+        }}
```
Comment (andrewor14) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: do you need double `}}` here?

**Comment by andrewor14** [comment]
Looks good!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala (hunk 1 above)
Comment (davies) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: I don't know, copied from ExecutionPage

Comment (zsxwing) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: It's necessary to escape `{` in xml.

**Comment by zsxwing** [comment]
LGTM

**Comment by davies** [comment]
@zsxwing Could you help me to fix the streaming test suite? It seems related to this PR.

**Comment by zsxwing** [comment]
retest this please

**Comment by davies** [comment]
Merging this into master, thanks!

**Pull request closed without merging**